pytest_split_tests/__init__.py

```diff
@@ -6,19 +6,23 @@
 from _pytest.config import create_terminal_writer
 import pytest
 
+def get_group_size_and_start(total_items, total_groups, group_id):
+    """Calculate group size and start index."""
+    base_size = total_items // total_groups
+    rem = total_items % total_groups
 
-def get_group_size(total_items, total_groups, group_id):
-    """Return the group size."""
-    base = total_items // total_groups
-    return base + 1 if group_id <= total_items % total_groups else base
+    start = base_size * (group_id - 1) + min(group_id - 1, rem)
+    size = base_size + 1 if group_id <= rem  else base_size
 
+    return (start, size)
 
-def get_group(items, group_size, group_id):
+def get_group(items, total_groups, group_id):
     """Get the items from the passed in group based on group size."""
-    start = group_size * (group_id - 1)
-    end = start + group_size
+    if not 0 < group_id <= total_groups:
+        raise ValueError("Invalid test-group argument")
 
-    return items[start:end]
+    start, size = get_group_size_and_start(len(items), total_groups, group_id)
+    return items[start:start+size]
```

Comment:
You could replace all of this with:
```
return items[group_id:len(items):group_size]
```
This is the three-argument slice-operator: `[start:end:step]`.